lib/hooks/use-order-quote-polling.ts

```diff
@@ -0,0 +1,132 @@
+import type { OrderQuote } from "@tscircuit/fake-snippets/schema"
+import { registryKy } from "lib/utils/get-registry-ky"
+import { useEffect, useRef, useState } from "react"
+import { useCallback } from "react"
+
+type PollingState = {
+  status: "idle" | "loading" | "success" | "error"
+  data: OrderQuote | null
+  error: Error | null
+}
+
+export const useOrderQuotePolling = (packageReleaseId: string) => {
+  const [state, setState] = useState<PollingState>({
+    status: "idle",
+    data: null,
+    error: null,
+  })
+  const abortControllerRef = useRef<AbortController | null>(null)
+
+  const createOrderQuotes = useCallback(async () => {
+    const { order_quote_id } = await registryKy
+      .post("order_quotes/create", {
+        json: {
+          package_release_id: packageReleaseId,
+          vendor_name: "jlcpcb", // TODO: Remove this once we have more vendor
+        },
+      })
+      .json<{ order_quote_id: string }>()
+    return order_quote_id
+  }, [packageReleaseId])
```

Comment:
if the function name is `useOrderQuotePolling`, i don't think it should create orders

Reply:
yeah, will separate that out